language/documentation/examples/math-puzzle/sources/puzzle.move

```diff
@@ -0,0 +1,33 @@
+module 0x42::Puzzle {
```

Comment:
Cute example.  It would be fun to do other puzzles in ways that illustrate points we want to teach about the spec lang or the prover.
One variation might be to put the variables in a vector of length 8, then say forall x in v: 1 <= x && x <= 9

Reply:
Yeah, puzzles and games are fun and seems to be approachable.

Good point! I thought about using a vector, but was not sure whether I needed a loop invariant for that. Perhaps, the new macro feature may help.